src/process/mod.rs

```diff
@@ -137,10 +137,10 @@ impl Process {
 
         let stack = OneMbStack::new()?;
         let mut process = AsyncWormhole::new(stack, move |yielder| {
-            let yielder_ptr = &yielder as *const AsyncYielderCast as usize;
+            let yielder_ptr = &yielder as *const AsyncYielder<Result<A::Return>> as usize;
```

Comment:
We are casting the yielder back to `AsyncYielder<Result<()>>` here: https://github.com/lunatic-solutions/lunatic/blob/unify_state_wrapper/src/process/mod.rs#L67

This should also be updated to use the new type. I forgot why I couldn't just pass around the pointer as `*const AsyncYielder<Result<A::Return>>` and needed to cast it to `usize`. Maybe this could be refactored a bit so that we don't destroy the type information.